neptune-core/src/application/json_rpc/server/service.rs

```diff
@@ -89,17 +99,183 @@ impl RpcApi for RpcServer {
                 .collect(),
         }
     }
+
+    async fn get_block_digest_call(
+        &self,
+        request: GetBlockDigestRequest,
+    ) -> GetBlockDigestResponse {
+        let state = self.state.lock_guard().await;
+
+        GetBlockDigestResponse {
+            digest: request.selector.as_digest(&state).await,
+        }
+    }
+
+    async fn get_block_call(&self, request: GetBlockRequest) -> GetBlockResponse {
+        let state = self.state.lock_guard().await;
+        let archival_state = state.chain.archival_state();
+
+        let block = match request.selector.as_digest(&state).await {
```

Comment:
Not super happy with these bindings here. They make lock acquisition a bit hard to reason about for me. Can you maybe drop the binding of `archival_state`?